databricks_cli/configure/config.py

```diff
@@ -47,9 +48,19 @@ def decorator(*args, **kwargs):
     return decorator
 
 
+def get_profile_from_context():
+    ctx = click.get_current_context()
+    context_object = ctx.ensure_object(ContextObject)
+    return context_object.get_profile()
+
+
 def profile_option(f):
-    return click.option('--profile', required=False, default=DEFAULT_SECTION,
-                        help='CLI connection profile to use.')(f)
+    def callback(ctx, param, value): #  NOQA
+        if value is not None:
+            context_object = ctx.ensure_object(ContextObject)
+            context_object.set_profile(value)
+    return click.option('--profile', required=False, default=None, callback=callback,
+                        expose_value=False, help='CLI connection profile to use.')(f)
```

Comment:
Should mention the default value in the help message. Seems not displayed previously either.